app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java

```diff
@@ -20,4 +34,49 @@ public static boolean putEventStringsInData(Map<String, String> event, Map<Strin
         }
         return false;
     }
+
+    //<-------------------¦ Preferences ¦--------------------------------------------------->
+
+    public static final String SHARED_PREFS = "com.github.houseorganizer.houseorganizer.sharedPrefs";
+
+    public static SharedPreferences getSharedPrefs(Activity a) {
+        return a.getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
+    }
+
+    public static SharedPreferences.Editor getSharedPrefsEditor(Activity a) {
+        SharedPreferences prefs = getSharedPrefs(a);
+        return prefs.edit();
+    }
+
+    //<-------------------¦ Messages ¦--------------------------------------------------->
+
+    public enum ErrorType {
+        INVALID_PASSWORD,
+        INVALID_EMAIL,
+        EMAIL_USED,
+        INPUTS_EMPTY
+    }
+
+    public static void logAndToast(List<String> logTagAndMsg, Exception e, Context cx, String toastMsg) {
+        assert logTagAndMsg.size() == 2;
+
+        Log.w(logTagAndMsg.get(0), logTagAndMsg.get(1), e);
+        Toast.makeText(cx, toastMsg, Toast.LENGTH_SHORT).show();
+    }
+
+    public static void displayErrorMessage(ErrorType err, TextView error_field) {
+        switch (err) {
+            case INVALID_EMAIL:
+                error_field.setText(R.string.email_not_valid);
+                break;
+            case INVALID_PASSWORD:
+                error_field.setText(R.string.password_not_valid);
+                break;
+            case EMAIL_USED:
+                error_field.setText(R.string.email_already_used);
+                break;
+            case INPUTS_EMPTY:
+                error_field.setText(R.string.inputs_empty);
```

Comment:
Im also not too sure about extracting this method since its one line and the line is different in every case

Reply:
Yeah I guess its not very useful but it does make the email login methods a tiny bit clearer. It was already extracted to LoginHelpers, I just moved it to Util to make it more generic.